Corrected detail view error. See #4 (Update to the new Cursive 0.10 API

gbonnema opened the pull request:
See my comment on issue #4. Needed to have ID directly on TextView for it to be recognized.,

gbonnema committed:
Corrected detail view error. See #4 (Update to the new Cursive 0.10 API).

gyscos commented on src/bin/tupm/ui.rs:
```diff
@@ -691,8 +691,8 @@ impl Ui {
-            let mut detail = match s.find_id::<TextView>(VIEW_ID_DETAIL) {
-                Some(x) => x,
+            let mut detail: ViewRef<TextView> = match s.find_id(VIEW_ID_DETAIL) {
+                Some(acc_det) => acc_det,
```
You could also write this:

```rust
s.call_on_id(VIEW_ID_DETAIL, |detail: &mut TextView| {
    detail.set_content(render_account_text(account, false));
    ui_tx_clone.send(UiMessage::UpdateStatus).unwrap();
});
```

gbonnema replied:
Right! Thank you for the review! The result certainly is very elegant. I included it into the PR and pushed the result (after testing ofcourse).

gbonnema replied:
P.S. Not sure about the button "Resolve Conversation" below. For whom is that meant?

gbonnema committed:
Review on PR #6 by Gyscos.

simmons committed:
Minor code cleanup

1. Remove .ackrc (Local concerns don't need to be committed to the
repository.)
2. Now that we are using the updated reqwest crate, remove the unused
old code for manually constructing a multipart body, along with the
associated comments and crates.
3. Make "application/octet-stream" a constant. (Avoid magic values.)
4. With the new Cursive API, AccountSelectView::selection() can just
return an Option directly from self.content.selection() -- no need for a
condition to check Some/None.

simmons committed:
Cargo update (including fixed ncurses)

Among other things, this upgrades the ncurses dependency (via Cursive)
to 5.99.0, which includes the fix for Unicode output.

simmons committed:
Revert "Corrected bogus lines by always requesting a password. Cause of the problem is unknown."

This reverts commit d1285bc44ab5c47ed6e1ab14be5a9cf718610b88.

simmons committed:
Fix: Don't exit in library code.

The upm create (as opposed to tupm) is a library which may potentially
be used by other programs in a variety of environments. (For example,
GUI programs, web application backends, etc.) Thus, it should not exit
or print to standard error, but simply return an error to the caller
when such a condition occurs.

simmons commented:
Hi Guus,

I'm merging this into the `fix_incompatibilities` branch, along with a few extra commits to perform some minor cleanup and revert the "always request a password" change. (The ncurses crate is fixed now, so there's no need for a workaround.) I also tweaked the reqwest instantiation in the upm crate to return an error instead of printing to stderr and exiting, since other developers who might use the library wouldn't expect it to exit on error.

If all looks well, in a few days I'll merge everything into master and update crates.io.

Thanks for the commits! It will be handy to have tupm work with the APIs of current dependency versions.

David
